File: PowerBiPlayground/Embedded/EmbedConfig.cs
namespace PowerBiPlayground.Embedded
{
    public class EmbedConfig
    {
        public string ReportId { get; }

        public string EmbedUrl { get; }

        public string Token { get; }

        public EmbedConfig(string reportId, string embedUrl, string token)
        {
            this.ReportId = reportId;
            this.EmbedUrl = embedUrl;
            this.Token = token;
        }
    }
}

File: PowerBiPlayground/ClientApp/src/app/powerbi/embed-config.ts
import { required, dataMember } from 'santee-dcts';
import { Injectable } from '@angular/core';

export class EmbedConfig {
    @dataMember()
    @required()
    public reportId: string;

    @dataMember()
    @required()
    public embedUrl: string;

    @dataMember()
    @required()
    public token: string;
}
